### ISSUE TITLE
🐛 BUG: Markdown: generated heading ids are based on raw source text, not final output

### ISSUE THREAD

**Issue by natemoo-re** [snowpackjs/astro#300]
For example, the following code 

```
---
const content = 'world';
---
<Markdown># Hello {content}</Markdown>
```

generates 

```
<h1 id="hello-content">Hello world</h1>
```

when it should generate
```
<h1 id="hello-world">Hello world</h1>
```

**Comment by jasikpark** [comment]
> This seems like a tough one given that we generate Markdown at compile time. I suppose moving it to runtime would fix it. Is it worth it to fix this issue though?

Well, shouldn't it be a case of whether it's prerendered or not? So in the example given, it's possible to add correctly named id's after all of the compilation steps, and there's no way to correctly name runtime component id's until they run, so they might as well default to the node text content and then possibly get rewritten on load?

**Comment by matthewp** [comment]
@jasikpark yeah, you're absolutely right, we *could* injection id calculation code into the runtime. Would still be non-trivial but would sidestep the problem of moving all markdown compilation to runtime (which I would be hesitant to do for perf reasons).

**Comment by jasikpark** [comment]
I think maybe this is a case where we should move from a `remark` plugin to a `rehype` plugin for the heading id's? 

https://github.com/remarkjs/remark-slug v.s. https://github.com/rehypejs/rehype-slug 

<img width="823" alt="image" src="https://user-images.githubusercontent.com/10626596/132590290-68d932fe-f56a-4458-aea3-70e3354e2f50.png">

**Comment by jasikpark** [comment]
Though maybe this is something that should wait for the new compiler 🤔

**Comment by jonathantneal** [comment]
This appears to be working in the new compiler.

https://stackblitz.com/edit/github-unkrm4?file=src%2Fpages%2Findex.astro

**Comment by natemoo-re** [comment]
Following up on Jonathan's comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
